src/rapids_pre_commit_hooks/copyright.py

```diff
@@ -184,31 +184,39 @@ def get_target_branch_upstream_commit(repo, args):
         except ValueError:
             return None
 
+    commits_to_try = []
     try:
         target_branch = repo.heads[target_branch_name]
     except IndexError:
-        target_branch = None
-    if target_branch:
-        target_branch_upstream = target_branch.tracking_branch()
-        if target_branch_upstream:
-            return target_branch_upstream.commit
 
-    def try_get_ref(remote):
+        def try_get_ref(remote):
+            try:
+                return remote.refs[target_branch_name]
+            except IndexError:
+                return None
+
         try:
-            return remote.refs[target_branch_name]
-        except IndexError:
-            return None
+            upstream_commit = max(
+                (
+                    upstream
+                    for remote in repo.remotes
+                    if (upstream := try_get_ref(remote))
```

Comment:
Does whatever type `remote.refs` is support `.get()`? If it does, then this inner `try_get_ref()` function could be removed in favor of this:

```python
if (upstream := remote.refs.get(target_branch_name, None))
```

I think that's be a bit easier to follow. Especially since `try_get_ref` is reaching back up into the calling environment to find `target_branch_name` instead of being passed it.

Reply:
`remote.refs` is an instance of `git.util.IterableList`, which does not support `.get()`.

Reply:
alright thanks for confirming